python/common/org/python/exceptions/BaseException.java

```diff
@@ -4,18 +4,32 @@
         __doc__ = "Common base class for all exceptions"
 )
 public class BaseException extends org.python.types.Object {
+    @org.python.Attribute()
+    org.python.types.Tuple args = new org.python.types.Tuple();
+
     public BaseException() {
         super();
-        // System.out.println("EX: " + this);
     }
 
     public BaseException(String msg) {
         super(msg);
-        // System.out.println("EX: " + this);
+        this.args.value.add(new org.python.types.Str(msg));
     }
 
     public BaseException(org.python.Object[] args, java.util.Map<java.lang.String, org.python.Object> kwargs) {
-        super(args[0].toString());
+        super(buildMessage(buildTuple(args)));
+        this.args = buildTuple(args);
+    }
+
+    private static String buildMessage(org.python.types.Tuple tupArgs) {
```

Comment:
Note: these static methods were needed because we need these to execute before `super()`

Reply:
Heh - clever. It occurs to me that this is a generic pattern that would work for any "run this code before super()" situation.